ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -957,6 +1043,163 @@ def wait_until_cpu_port_up(self):
+                    # This is a sent packet - keep track of it as payload_id:timestamp.
+                    sent_payload = int(str(packet[scapyall.UDP].payload))
+                    sent_packets[sent_payload] = packet.time
+                    continue
+                if packet[scapyall.Ether].src == self.dut_mac:
+                    # This is a received packet.
+                    received_time = packet.time
+                    received_payload = int(str(packet[scapyall.UDP].payload))
+                    received_counter += 1
+                if not (received_payload and received_time):
+                    # This is the first valid received packet.
+                    prev_payload = received_payload
+                    prev_time = received_time
+                    continue
+                if received_payload - prev_payload > 1:
+                    # Packets in a row are missing, a disruption.
+                    lost_id = (received_payload -1) - prev_payload # How many packets lost in a row.
+                    disrupt = (sent_packets[received_payload] - sent_packets[prev_payload + 1]) # How long disrupt lasted.
+                    # Add disrupt to the dict:
+                    self.lost_packets[prev_payload] = (lost_id, disrupt, received_time - disrupt, received_time)
+                    self.log("Disruption between packet ID %s and %s. For %s " % (prev_payload, received_payload, disrupt))
+                    if not self.disruption_start:
+                        self.disruption_start = datetime.datetime.fromtimestamp(prev_time)
+                    self.disruption_stop = datetime.datetime.fromtimestamp(received_time)
+                prev_payload = received_payload
+                prev_time = received_time
+        else:
+            self.log("Packets list is empty")
+        if self.lost_packets:
+            self.disrupts_count = len(self.lost_packets) # Total disrupt counter.
+            # Find the longest loss with the longest time:
+            max_disrupt_from_id, (self.max_lost_id, self.max_disrupt_time, self.no_routing_start, self.no_routing_stop) = \
+                max(self.lost_packets.items(), key = lambda item:item[1][0:2])
+            self.total_disrupt_packets = sum([item[0] for item in self.lost_packets.values()])
+            self.total_disrupt_time = sum([item[1] for item in self.lost_packets.values()])
+            self.log("Disruptions happen between %s and %s after the reboot." % \
+                (str(self.disruption_start - self.reboot_start), str(self.disruption_stop - self.reboot_start)))
+        else:
+            self.log("Gaps in forwarding not found.")
+        self.log("Total incoming packets captured %s" % str(received_counter))
+        if packets:
+            filename = '/tmp/capture_filtered.pcap'
+            scapyall.wrpcap(filename, packets)
+            self.log("Filtered pcap dumped to %s" % filename)
+        else:
+            self.log("Filtered Pcap file is empty.")
```

Comment:
Should this be a failure?

Reply:
Added the failure before the analyzer:
self.assertTrue(packets, "Sniffer failed to capture any traffic")

Added the faulure after the anayzer:
self.assertTrue(received_counter, "Sniffer failed to filter any traffic from DUT")

And removed redundant 'else' clauses.